saleor/graphql/menu/mutations.py

```diff
@@ -213,21 +209,24 @@ class Meta:
         description = 'Assigns storefront\'s navigation menus.'
 
     @classmethod
-    @permission_required(['menu.manage_menus', 'site.manage_settings'])
-    def mutate(cls, root, info, navigation_type, menu=None):
-        errors = []
+    def user_is_allowed(cls, instance, input):
+        return instance.has_perms([
+            'menu.manage_menus', 'site.manage_settings'])
+
+    @classmethod
+    def perform_mutation(cls, root, info, navigation_type, menu=None):
         site_settings = info.context.site.settings
         if menu is not None:
-            menu = cls.get_node_or_error(
-                info, menu, errors=errors, field='menu')
-        if not errors:
-            if navigation_type == NavigationType.MAIN:
-                site_settings.top_menu = menu
-                site_settings.save(update_fields=['top_menu'])
-            elif navigation_type == NavigationType.SECONDARY:
-                site_settings.bottom_menu = menu
-                site_settings.save(update_fields=['bottom_menu'])
-            else:
-                raise AssertionError(
-                    'Unknown navigation type: %s' % navigation_type)
-        return AssignNavigation(menu=menu, errors=errors)
+            menu = cls.get_node_or_error(info, menu, field='menu')
+
+        if navigation_type == NavigationType.MAIN:
+            site_settings.top_menu = menu
+            site_settings.save(update_fields=['top_menu'])
+        elif navigation_type == NavigationType.SECONDARY:
+            site_settings.bottom_menu = menu
+            site_settings.save(update_fields=['bottom_menu'])
+        else:
+            raise AssertionError(
```

Comment:
Why `AssertionError`? And again I see we use enums, so technically this code should never be executed.